src/runtime/graph/graph_runtime.cc

```diff
@@ -103,27 +103,74 @@ class GraphRuntime : public ModuleNode {
   void SetInput(int index, DLTensor* data_in) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(data_in, &data_entry_[eid], nullptr));
+    data_entry_[eid].CopyFrom(data_in);
+  }
+  /*!
+   * \brief Get the number of outputs
+   *
+   * \return The number of outputs from graph.
+   */
+  int GetNumOutputs(void) {
+    return outputs_.size();
+  }
+  /*!
+   * \brief Return NDArray for given input index.
+   * \param index The input index.
+   *
+   * \return NDArray corresponding to given input node index.
+   */
+  NDArray GetInputAsNDArray(int index) {
+    CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
+    uint32_t eid = this->entry_id(input_nodes_[index], 0);
+    return NDArray(data_entry_[eid]);
+  }
+  /*!
+   * \brief Return NDArray for given output index.
+   * \param index The output index.
+   *
+   * \return NDArray corresponding to given output node index.
+   */
+  NDArray GetOutputAsNDArray(int index) {
+    CHECK_LT(static_cast<size_t>(index), outputs_.size());
+    uint32_t eid = this->entry_id(outputs_[index]);
+    tvm::runtime::DeviceAPI::Get(ctx_)->StreamSync(ctx_, nullptr);
+    return NDArray(data_entry_[eid]);
   }
   /*!
    * \brief Copy index-th input to data_out
    * \param index The input index.
    * \param data_out The output
    */
-  void GetInput(int index, DLTensor* data_out) {
+  void GetInputAsDLTensor(int index, DLTensor* data_out) {
```

Comment:
Remove this function, as it is not performance critical. The history get_input with copy can simply done with get_input(index).CopyTo(data_out) in python side